mathesar_ui/src/utils/permissions.ts

```diff
@@ -0,0 +1,20 @@
+import type { UserRole } from '@mathesar/api/users';
+import { MissingExhaustiveConditionError } from '@mathesar/utils/errors';
+
+export type AccessOperation = 'modifyData' | 'performCrud' | 'modifyAccess';
```

Comment:
Fixed in https://github.com/centerofci/mathesar/pull/2443/commits/52b8be8ea38dae298e2d76055c0182171980ed0f.
Comments added in https://github.com/centerofci/mathesar/pull/2443/commits/558da18ebea2c019c14f8ae546c11e11fd218219.

The names are changed based on the common description i.e.
```typescript
/**
 * - `canEditTableRecords`: DML operations on table rows and cells.
 * - `canEditMetadata`: CRUD operations on data stored on our internal database.
 *   - This includes column display options, record summary etc.,
 *   - This also includes all operations on Explorations.
 * - `canExecuteDDL`: DDL operations on objects stored directly on user database,
 *   including schemas, tables, columns, constraints etc.,
 *   This also includes:
 *    - Descriptions for schemas and tables: Because these are stored
 *    in the user database.
 *    - Importing data: Because this creates a new table.
 * - `editPermissions`: Changes to user access.
 */
export type AccessOperation =
  | 'canEditTableRecords'
  | 'canEditMetadata'
  | 'canExecuteDDL'
  | 'canEditPermissions';
```

I used `canExecuteDDL` instead of `canEditSchemaStructure`, because schema is rather an ambiguous term, it could refer to the schema object or the db schemata. `canExecuteDDL` is more clear.